relay-server/src/services/project/metrics/mod.rs

```diff
@@ -0,0 +1,99 @@
+use std::marker::PhantomData;
+use std::ops::Deref;
+
+use relay_metrics::aggregator::AggregatorBuckets;
+use relay_metrics::Bucket;
+
+mod post;
+mod pre;
+
+pub use self::post::PostProjectTransform;
+pub use self::pre::PreProjectTransform;
+
+/// Transforms [`Buckets`] into different [`Buckets`] applying pre-defined transforms.
+pub trait Transformer<S> {
+    /// The new state the buckets will be transformed into.
+    type NewState;
```

Comment:
Should we rename `S` and `NewState` to something like `StateIn` `StateOut`, or `InputState` `OutputState`?